org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacBindingResolver.java

```diff
@@ -175,8 +175,8 @@ ITypeBinding resolveType(EnumDeclaration enumDecl) {
 	public IBinding getBinding(final Symbol owner, final com.sun.tools.javac.code.Type type) {
 		if (owner instanceof final PackageSymbol other) {
 			return new JavacPackageBinding(other, this);
-		} else if (owner instanceof TypeSymbol) {
-			return new JavacTypeBinding(type, this);
+		} else if (owner instanceof TypeSymbol typeSymbol) {
+			return new JavacTypeBinding(typeSymbol.type, this);
```

Comment:
OK, but the issue is that the type we receive from this method doesn't really match the typeSymbol, it returns something that seems unrelated. This can be reproduced when clicking Ctrl+click in a method body over a parameter.